lib/gollum-lib/wiki.rb

```diff
@@ -504,31 +504,14 @@ def size(ref = nil)
     # query - The string to search for
     #
     # Returns an Array with Objects of page name and count of matches
-    def search(query)
+    def search(query, return_terms = false)
       options = {:path => page_file_dir, :ref => ref}
-      results = {}
-      @repo.git.grep(query, options).each do |hit|
-        name = hit[:name]
-        count = hit[:count]
-        # Remove ext only from known extensions.
-        # test.pdf => test.pdf, test.md => test
-        file_name = Page::valid_page_name?(name) ? name.chomp(::File.extname(name)) : name
-        results[file_name] = count.to_i
-      end
-
-      # Use ls_files '*query*' to search for file names. Grep only searches file content.
```

Comment:
Is this reasoning no longer valid? Using `ls_files` to search for file names?

Reply:
No, when looping through the repo the adapter already has the filenames, so this additional loop through the repo is unnecessary and expensive. Ergo we'll be checking whether the filename is a hit in the adapter (I'll update the adapter spec suite to reflect this -- in fact I think most tests that are now included here should move to the adapter spec suite).